File: editorial_ch/chap4-6/prob4-6-2.java
import java.util.*;

class Main {
	public static void main(String[] args) {
		// 輸入
		Scanner sc = new Scanner(System.in);
		int X = sc.nextInt();
		int Y = sc.nextInt();
		
		// 狀況區分（a, b 會變成負的時候）
		if ((2 * Y - X) < 0 || (2 * X - Y) < 0) {
			System.out.println("0");
		}
		
		// 狀況區分（a, b 不會成為整數的時候）
		else if ((2 * Y - X) % 3 != 0 || (2 * X - Y) % 3 != 0) {
			System.out.println("0");
		}
		
		// 答案不為零時
		else {
			long MOD = 1000000007;
			long bunshi = 1, bunbo = 1;
			long a = (2 * Y - X) / 3;
			long b = (2 * X - Y) / 3;
			
			// 求出二項係數的分子與分母（步驟 1.／步驟 2.）
			for (int i = 1; i <= a + b; i++) {
				bunshi = (bunshi * i) % MOD;
			}
			for (int i = 1; i <= a; i++) {
				bunbo = (bunbo * i) % MOD;
			}
			for (int i = 1; i <= b; i++) {
				bunbo = (bunbo * i) % MOD;
			}
			
			// 求出答案（步驟 3.）
			long answer = division(bunshi, bunbo, MOD);
			System.out.println(answer);
		}
	}
	static long modpow(long a, long b, long m) {
		// 重複平方法（p 取如 a^1, a^2, a^4, a^8, ... 的值）
		long p = a, answer = 1;
		for (int i = 0; i < 30; i++) {
			if ((b & (1 << i)) != 0) {
				answer = (answer * p) % m;
			}
			p = (p * p) % m;
		}
		return answer;
	}
	static long division(long a, long b, long m) {
		// division(a, b, m) 為傳回 a÷b mod m 的函式
		return (a * modpow(b, m - 2, m)) % m;
	}
}

File: editorial_ch/chap4-6/prob4-6-2.py
# 重複平方法（p 取如 a^1, a^2, a^4, a^8, ... 的值）
def modpow(a, b, m):
	p = a
	answer = 1
	for i in range(30):
		if (b & (1 << i)) != 0:
			answer = (answer * p) % m
		p = (p * p) % m
	return answer

# division(a, b, m) 為傳回 a÷b mod m 的函式
def division(a, b, m):
	return (a * modpow(b, m - 2, m)) % m

# 輸入
X, Y = map(int, input().split())
mod = 1000000007

# 狀況區分（a, b 會變成負的時候）
if (2 * Y - X) < 0 or (2 * X - Y) < 0:
	print(0)

# 狀況區分（a, b 不會成為整數的時候）
elif (2 * Y - X) % 3 != 0 or (2 * X - Y) % 3 != 0:
	print(0)

# 答案不為零的案例
else:
	bunshi = 1
	bunbo = 1
	a = (2 * Y - X) // 3
	b = (2 * X - Y) // 3
	
	# 求出二項係數的分子與分母（步驟 1. / 步驟 2.）
	for i in range(1, a + b + 1):
		bunshi *= i
		bunshi %= mod
	for i in range(1, a + 1):
		bunbo *= i
		bunbo %= mod
	for i in range(1, b + 1):
		bunbo *= i
		bunbo %= mod
	
	# 求出答案（步驟 3.）
	print(division(bunshi, bunbo, mod))